### ISSUE TITLE
Encoding the email returns unwanted characters

### ISSUE THREAD

**Issue by christinamcx** [mailcheck/mailcheck#155]
I have entered this email: 
petros$%$^%%@gmaill.com 
and mailchecker returned this suggestion: 
petros$%$^%25%25@gmail.com

The problem is in line 224 of mailcheck.js where the replace after the encode does not replace all. This should be modified to do a replace all. For example for replacing all occurrences of %25, it should be .replace(/%25/g, '%') 

Can you please look into it?

**Comment by ferreiro** [comment]
@christinamcx good catch! I fixed it here: https://github.com/ZooTools/email-spell-checker/pull/4 

Let me know if you have any feedback.

---

Disclaimer: This library is no longer maintained.

I rewrote it entirely and update it for 2022 and onward: https://github.com/ZooTools/email-spell-checker

- 💙 Written in TypeScript and removed jQuery
- ✅ Fixed a couple of bugs like [this](https://github.com/ZooTools/email-spell-checker/pull/3) or [this](https://github.com/ZooTools/email-spell-checker/pull/4)
- 🚀 Reduced bundle size to <2KB. 
- ✨ Update TLDs (69+) and added modern startup domains (like .io, .so, .xyz or .dev)
- 🙏 Implemented suggestions that people made in this repo that were ignored.

Link: https://github.com/ZooTools/email-spell-checker

Come check it out and give it a ⭐️ for the effort.

**Comment by skyfallsin** [comment]
Fixed on master in e5bca78: encodeEmail() now restores every literal percent sign rather than only the first one, with regression coverage for repeated percent characters.
